src/application/scene/Todo.java

```diff
@@ -1,14 +1,186 @@
 package application.scene;
 
 import javafx.fxml.FXML;
-import javafx.scene.control.Button;
+import javafx.scene.control.*;
+
+import java.io.*;
+import java.time.LocalDate;
+import java.util.List;
 
 public class Todo {
-	  @FXML private Button button_dashboard;
+    @FXML
+    private Button button_dashboard;
+    @FXML
+    private Button button_addEvent;
+    @FXML
+    private Button button_moveDOING;
+    @FXML
+    private Button button_moveDONE;
+    @FXML
+    private Button button_clearDONE;
+    @FXML
+    private TextField textField_eventName;
+    @FXML
+    private DatePicker datePicker;
+    @FXML
+    private ListView<String> listView_TODO;
+    @FXML
+    private ListView<String> listView_DOING;
+    @FXML
+    private ListView<String> listView_DONE;
+    @FXML
+    private ChoiceBox<String> choiceBox_status;
+
+
+    public void initialize() {
+
+        textField_eventName.setPromptText("TODO Events");
+
+        datePicker.setValue(LocalDate.now());
+
+        //Add TODO DOING DONE status to the choiceBox dropdown
+        choiceBox_status.getItems().addAll("TODO", "DOING", "DONE");
+        choiceBox_status.setValue("TODO");
+
+        getHistory(listView_TODO, "TODO");
+        getHistory(listView_DOING, "DOING");
+        getHistory(listView_DONE, "DONE");
+
+
+    }
+
+    @FXML
+    private void addEvent() {
+        String status = choiceBox_status.getValue();
+        String eventMessage = datePicker.getValue() + " " + textField_eventName.getText();
+
+        if (textField_eventName.getText().isEmpty()) {
+            noContentAlert();
+        } else {
+            switch (status) {
+                case "TODO":
+                    addToList(listView_TODO, "TODO", eventMessage);
+                    break;
+                case "DOING":
+                    addToList(listView_DOING, "TODO", eventMessage);
+                    break;
+                case "DONE":
+                    addToList(listView_DONE, "TODO", eventMessage);
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+
```

Comment:
On these two move functions, i was getting lots of errors in my stacktrace when I tried to move an empty list. It didn't affect the frontend at all which is good, but still for best practice could an error catch be implemented?

Reply:
Patched, add exception handler